core/services/head_tracker.go

```diff
@@ -71,6 +69,13 @@ type HeadTracker struct {
 	subscriptionSucceeded chan struct{}
 }
 
+type RawHead struct {
+	Number     string
+	Hash       string
+	ParentHash string
+	Timestamp  string
+}
```

Comment:
I would add explicit JSON struct tags here.
It's not clear to me what an example JSON document would look like (is it `{"ParentHash":"..`, or `{"parent-hash":"...` or `{"parent_hash":"..`? It's the first one) and I think the tags would help!

Reply:
Isn't it clear that it would use the default?

GoLang's json library camelises by default I believe.

Reply:
Ok. I think this is just personal preference.